### PR TITLE
Fix cursor scroll on Home and End in interactive window

### PR DESCRIPTION
**Pull request by genlu**
Fix https://github.com/dotnet/roslyn/issues/5176

@amcasey @cston @KevinH-MS @tmat @tannergooding @ManishJayaswal

### CODE REVIEW

**Comment by cston** [comment]
LGTM

**Comment by ManishJayaswal** [comment]
:+1:

File: src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs [hunk 1]
```diff
@@ -349,6 +349,18 @@ public void CallEndOnNonUIThread()
             Task.Run(() => Window.Operations.End(true)).PumpingWait();
         }
 
+        public void ScrollToCursorOnHomeAndEndOnNonUIThread()
+        {
+            Window.InsertCode(new string('1', 512));    // a long input string 
+
+            Task.Run(() => Window.Operations.Home(false)).PumpingWait();
```
Comment (amcasey) [inline src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs]: I don't think you need ```Task.Run```.

File: src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs [hunk 2]
```diff
@@ -349,6 +349,18 @@ public void CallEndOnNonUIThread()
...
+            Assert.True(Window.TextView.TextViewModel.IsPointInVisualBuffer(Window.TextView.Caret.Position.BufferPosition,
```
Comment (amcasey) [inline src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs]: Consider extracting a local for ```Window.TextView```.

**Comment by amcasey** [comment]
:+1:

**Pull request merged**